sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -372,6 +372,14 @@ private[spark] object SQLConf {
           "possible, or you may get wrong result.",
     isPublic = false)
 
+  val CANONICAL_NATIVE_VIEW = booleanConf("spark.sql.nativeView.canonical",
+    defaultValue = Some(false),
```

Comment:
I'm for enabling canonical native view by default, but not sure whether native view should be enabled by default. After all, native view is still feature incomplete, no matter canonicalized or not.